### ISSUE TITLE
I can't get node-opencl installed

### ISSUE THREAD

**Issue by NCARalph** [mikeseven/node-opencl#34]
I'm attempting to install node-opencl on a Windows 10 system with Visual Studio 2015 without success. If there's another better place to report problems let me know.

I've installed 
The current Intel OpenCL development environment, 
Python 2.7 and created a PYTHON environment variable pointing to the exe, 
the current node.js,
node-gyp with the GYP_MSVS_VERSION=2015 variable,
but cannot get npm to install node-opencl.

This is the error stream I get back from an administrative command prompt:
C:\WINDOWS\system32>npm install node-opencl

> memwatch-next@0.2.10 install C:\WINDOWS\system32\node_modules\memwatch-next
> node-gyp rebuild

C:\WINDOWS\system32\node_modules\memwatch-next>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild )
MSBUILD : error MSB1009: Project file does not exist.
Switch: build/binding.sln
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:88:13)
gyp ERR! stack     at ChildProcess.emit (events.js:173:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:201:12)
gyp ERR! System Windows_NT 10.0.10240
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\WINDOWS\system32\node_modules\memwatch-next
gyp ERR! node -v v5.2.0
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok
npm WARN install:memwatch-next@0.2.10 memwatch-next@0.2.10 install: `node-gyp rebuild`
npm WARN install:memwatch-next@0.2.10 Exit status 1
npm WARN ENOENT ENOENT: no such file or directory, open 'C:\WINDOWS\system32\package.json'
npm WARN EPACKAGEJSON system32 No description
npm WARN EPACKAGEJSON system32 No repository field.
npm WARN EPACKAGEJSON system32 No README data
npm WARN EPACKAGEJSON system32 No license field.

**Comment by mikeseven** [comment]
It is a bug with memwatch-next apparently.

memwatch-next is used only in a test to track down memory leaks. It’s not used by node-opencl itself.

Just remove memwatch-next from dependencies in package.son and let me know. If this solves the install, I’ll move it to devDependencies or remove it.


On Dec 11, 2015, at 4:50 PM, NCARalph <notifications@github.com<mailto:notifications@github.com>> wrote:


I'm attempting to install node-opencl on a Windows 10 system with Visual Studio 2015 without success If there's another better place to report problems let me know

I've installed
The current Intel OpenCL development environment,
Python 27 and created a PYTHON environment variable pointing to the exe,
the current nodejs,
node-gyp with the GYP_MSVS_VERSION=2015 variable,
but cannot get npm to install node-opencl

This is the error stream I get back from an administrative command prompt:
C:\WINDOWS\system32>npm install node-opencl

memwatch-next@0210 install C:\WINDOWS\system32\node_modules\memwatch-next
node-gyp rebuild

C:\WINDOWS\system32\node_modules\memwatch-next>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\node_modules\node-gyp\bin\node-gypjs" rebuild ) else (node rebuild )
MSBUILD : error MSB1009: Project file does not exist
Switch: build/bindingsln
gyp ERR! build error
gyp ERR! stack Error: C:\Program Files (x86)\MSBuild\140\bin\msbuildexe failed with exit code: 1
gyp ERR! stack at ChildProcessonExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\buildjs:270:23)
gyp ERR! stack at emitTwo (eventsjs:88:13)
gyp ERR! stack at ChildProcessemit (eventsjs:173:7)
gyp ERR! stack at ProcessChildProcess_handleonexit (internal/child_processjs:201:12)
gyp ERR! System Windows_NT 10010240
gyp ERR! command "C:\Program Files\nodejs\nodeexe" "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gypjs" "rebuild"
gyp ERR! cwd C:\WINDOWS\system32\node_modules\memwatch-next
gyp ERR! node -v v520
gyp ERR! node-gyp -v v303
gyp ERR! not ok
npm WARN install:memwatch-next@0210 memwatch-next@0210 install: node-gyp rebuild
npm WARN install:memwatch-next@0210 Exit status 1
npm WARN ENOENT ENOENT: no such file or directory, open 'C:\WINDOWS\system32\packagejson'
npm WARN EPACKAGEJSON system32 No description
npm WARN EPACKAGEJSON system32 No repository field
npm WARN EPACKAGEJSON system32 No README data
npm WARN EPACKAGEJSON system32 No license field

**Comment by NCARalph** [comment]
I'm not sure what has changed, but now I'm getting a lot of errors from the gyp build with or without memwatch-next as a dependency. This data is with memwatch-next removed from the project.son file.

GYP_MSVS_VERSION=2015 is defined, but it looks like it's using auto as the Visual Studio version, but I don't know if that's a problem.

The install.txt file is the command output which looks like there's a problem in the types.h file.
I also included the npm-debug.log file
[install.txt](https://github.com/mikeseven/node-opencl/files/60283/install.txt)
[npm-debug.txt](https://github.com/mikeseven/node-opencl/files/60284/npm-debug.txt)

I've got the Intel OpenCL SDK installed, here's part of the path:
Path=C:\Program Files (x86)\Intel\OpenCL SDK\5.3\bin\x64

**Comment by mikeseven** [comment]
Just pushed the fix, tested on windows 10 x64 with msvs2015, mac clang and Linux gcc 5.

I'm not quite sure how we missed it. There were 2 code errors: one is an issue with msvs compiler, the second one was svm.cpp not updated.

Thanks for spotting it.

Let me know if it works for you and close the bug.

--mike



On Sat, Dec 12, 2015 at 2:49 PM -0800, "NCARalph" <notifications@github.com<mailto:notifications@github.com>> wrote:


I'm not sure what has changed, but now I'm getting a lot of errors from the gyp build with or without memwatch-next as a dependency. This data is with memwatch-next removed from the project.son file.

GYP_MSVS_VERSION=2015 is defined, but it looks like it's using auto as the Visual Studio version, but I don't know if that's a problem.

The install.txt file is the command output which looks like there's a problem in the types.h file.
I also included the npm-debug.log file
install.txt<https://github.com/mikeseven/node-opencl/files/60283/install.txt>
npm-debug.txt<https://github.com/mikeseven/node-opencl/files/60284/npm-debug.txt>

I've got the Intel OpenCL SDK installed, here's part of the path:
Path=C:\Program Files (x86)\Intel\OpenCL SDK\5.3\bin\x64

-
